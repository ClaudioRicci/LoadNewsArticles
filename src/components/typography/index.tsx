import styled from "styled-components";
import { colors, fonts, align, deviceSizes } from "../../themeVariables";

export const H1Tag = styled.h1`
  font: 900 1em/1rem ${fonts.header};
  color: ${colors.primary};
  text-align: ${align.left};
  border-bottom: 0.2rem ${colors.primary} solid;
  padding-bottom: 0.2rem;
  width: fit-content;
  grid-area: a;
  @media (max-width: ${deviceSizes.mobile}) {
    font: 900 1.4em/1.4rem ${fonts.header};
  }
`;

export const H2Tag = styled.h2`
  font: 900 0.8em/0.8rem ${fonts.header};
  color: ${colors.secondary};
  line-height: 1rem;
`;

export const ArticleTitle = styled.a`
  font: 900 0.84em/0.84rem ${fonts.header};
  color: ${colors.secondary};
  text-align: ${align.left};
  text-decoration: none;
  margin-bottom: 0.5rem;
  cursor: pointer;
`;

export const PTag = styled.p`
  line-height: 1rem;
`;
