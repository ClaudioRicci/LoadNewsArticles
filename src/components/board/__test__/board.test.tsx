import React from "react";
import ReactDOM from "react-dom";
import Board from "..";
import { H1Tag } from "../../typography";
import renderer from "react-test-renderer";
import "jest-styled-components";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

describe("Board Tests", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Board />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe("Title Tests", () => {
  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<H1Tag></H1Tag>, div);
  });

  test("H1 Renders Correctly", () => {
    const tree = renderer.create(<H1Tag />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("color", "#D9819D");
  });
});
