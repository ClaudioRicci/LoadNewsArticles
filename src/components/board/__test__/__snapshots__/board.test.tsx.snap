// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Tests H1 Renders Correctly 1`] = `
.c0 {
  font: 900 1em/1rem Verdana,Arial,Helvetica,sans-serif;
  color: #D9819D;
  text-align: left;
  border-bottom: 0.2rem #D9819D solid;
  padding-bottom: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  grid-area: a;
}

@media (max-width:768px) {
  .c0 {
    font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  }
}

<h1
  className="c0"
/>
`;
